plugins/pdr_deterministic_plugin/ufm_sim_web_service/pdr_algorithm.py

```diff
@@ -0,0 +1,557 @@
+                if symbol_ber_rate and symbol_ber_rate > threshold:
+                    self.logger.info(f"Isolation issue ({Constants.ISSUE_BER}) detected for port {port_obj.port_name} (speed: {port_obj.active_speed}, width: {port_obj.port_width}): "
+                                    f"symbol ber rate ({symbol_ber_rate}) is higher than threshold ({threshold})")
+                    return Issue(port_obj.port_name, Constants.ISSUE_BER)
+        return None
+
+    def read_next_set_of_high_ber_or_pdr_ports(self, ports_data, ports_counters):
+        """
+        Read the next set of ports and check if they have high BER, PDR, temperature or link downed issues
+        Receives: detected ports, isolated ports and telemetry data
+        Returns: lists of isolation issues
+        """
+        self.ports_data = ports_data
+
+        issues = {}
+        for _, row in ports_counters.iterrows():
+            port_name = f"{row.get(Constants.NODE_GUID, '').split('x')[-1]}_{row.get(Constants.PORT_NUMBER, '')}"
+            if self.exclude_list.contains(port_name):
+                # The port is excluded from analysis
+                continue
+            if self.test_mode:
+                # Adding the port data upon fetching the first telemetry data
+                if not self.ports_data.get(port_name):
+                    self.ports_data[port_name] = PortData(port_name=port_name,peer="0x"+port_name)
+            port_obj = self.ports_data.get(port_name)
+            if not port_obj:
+                if get_counter(Constants.RCV_PACKETS_COUNTER,row,0) == 0: # meaning it is down port
+                    continue
+                self.logger.warning("Port {0} not found in ports data".format(port_name))
+                continue
+            # Converting from micro seconds to seconds.
+            timestamp = get_timestamp_seconds(row)
+            #TODO add logs regarding the exact telemetry value leading to the decision
+            pdr_issue = self.check_pdr_issue(port_obj, row, timestamp)
+            temp_issue = self.check_temp_issue(port_obj, row, timestamp)
+            link_downed_issue = self.check_link_down_issue(port_obj, row, timestamp, ports_counters)
+            ber_issue = self.check_ber_issue(port_obj, row, timestamp)
+            port_obj.last_timestamp = timestamp
+            if pdr_issue:
+                issues[port_name] = pdr_issue
+            elif temp_issue:
+                issues[port_name] = temp_issue
+            elif link_downed_issue:
+                issues[port_name] = link_downed_issue
+            elif ber_issue:
+                issues[port_name] = ber_issue
+            # If out of operating conditions we'll overwrite the cause
+            if self.temp_check and self.is_out_of_operating_conf(port_name):
+                issues[port_name] = Constants.ISSUE_OONOC
+        return list(issues.values())
+
+    def should_deisolate(self, port_state):
```

Comment:
if this function is doing the port deisolation evaluation and the other function is performing the action then rename it to "eval_port_deisolation" and "perform_deisolation", also add a more informative docstring

Reply:
This function just checks if given port should be deisolated. It does not perform deisolation,

Reply:
More explanation added to docstring